noise-research/secure-transfer.md

```diff
@@ -0,0 +1,243 @@
+    - A commitment `H(sB||r)` for its static key `sB` where `r` is a random fixed-lenght value.
+
+2. The device `A`:
+
+    - scans the QR code;
+    - obtains `eB`, `contentTopicParams`, `messageNametag`, `Hash(sB|r)`;
+    - checks if `{application-name}` and `{application-version}` from `contentTopicParams` match the local application name and version: if not, aborts the pairing. 
+    - initializes the Noise handshake by passing `contentTopicParams`, `messageNametag` and `Hash(sB||r)` to the handshake prologue;
+    - executes the pre-handshake message, i.e. processes the key `eB`;
+    - executes the first handshake message over `contentTopic`, i.e. 
+        - processes and sends a Waku message containing an ephemeral key `eA`; 
+        - performs `DH(eA,eB)` (which computes a symmetric encryption key);
+        - attach as payload to the handshake message a commitment `H(sA|s)` for `A`'s static key `sA`, where `s` is a random fixed-lenght value;
+    - an 8-digits authorization code `authcode` obtained as `HKDF(h) mod 10^8` is displayed on the device, where `h`is the handshake value obtained once the first handshake message is processed.
+
+3. The device `B`:
+
+    - listens to messages sent to `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto` and locally filters only those with [Waku payload](https://rfc.vac.dev/spec/35/#abnf) starting with `messageNametag`. If any, continues.
+    - initializes the Noise handshake by passing `contentTopicParams`, `messageNametag` and `Hash(sB||r)` to the handshake prologue;
+    - executes the pre-handshake message, i.e. processes its static key `eB`;
+    - executes the first handshake message, i.e.
+        - obtains from the received message a public key `eA`. If `eA` is not a valid public key, the protocol is aborted.
+        - performs `DH(eA,eB)` (which computes a symmetric encryption key);
+        - decrypts the commitment `H(sA||s)` for `A`'s static key `sA`.
+    - an 8-digits authorization code `authcode` obtained as `HKDF(h) mod 10^8` is displayed on the device, where `h`is the handshake value obtained once the first handshake message is processed.
+
+4. Device `A` and `B` wait the user to confirm with an interaction (button press) that the authorization code displayed on both devices are the same. If not, the protocol is aborted.
+    
+5. The device `B`:
+
+    - executes the second handshake message, i.e.
+        - processes and sends his (encrypted) device static key `sB` over `contentTopic`;
+        - performs `DH(eA,sB)` (which updates the symmetric encryption key);
+        - attaches as payload the (encrypted) commitment randomness `r` used to compute `H(sB||r)`.
+
+6. The device `A`:
+
+    - listens to messages sent to `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto` and locally filters only those with Waku payload starting with `messageNametag`. If any, continues.
```

Comment:
Is there any specific reason you did not say ` listens to messages sent to  contentTopic` like in step 5 above?
```suggestion
    - listens to messages sent to `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto` and locally filters only those with Waku payload starting with `messageNametag`. If any, continues.
```